Engine/Module/ModuleDebugDraw.cpp

```diff
@@ -0,0 +1,619 @@
+                /* offset    = */ reinterpret_cast<void*>(offset));
+
+            checkGLError(__FILE__, __LINE__);
+
+            // Ditto.
+            glBindVertexArray(0);
+            glBindBuffer(GL_ARRAY_BUFFER, 0);
+        }
+    }
+
+    static GLuint handleToGL(dd::GlyphTextureHandle handle)
+    {
+        const std::size_t temp = reinterpret_cast<std::size_t>(handle);
+        return static_cast<GLuint>(temp);
+    }
+
+    static dd::GlyphTextureHandle GLToHandle(const GLuint id)
+    {
+        const std::size_t temp = static_cast<std::size_t>(id);
+        return reinterpret_cast<dd::GlyphTextureHandle>(temp);
+    }
+
+    static void checkGLError(const char* file, const int line)
+    {
+        GLenum err;
+        while ((err = glGetError()) != GL_NO_ERROR)
+        {
+            //errorF("%s(%d) : GL_CORE_ERROR=0x%X - %s", file, line, err, errorToString(err));
+        }
+    }
+
+    static void compileShader(const GLuint shader)
+    {
+        glCompileShader(shader);
+        checkGLError(__FILE__, __LINE__);
+
+        GLint status;
+        glGetShaderiv(shader, GL_COMPILE_STATUS, &status);
+        checkGLError(__FILE__, __LINE__);
+
+        if (status == GL_FALSE)
+        {
+            GLchar strInfoLog[1024] = { 0 };
+            glGetShaderInfoLog(shader, sizeof(strInfoLog) - 1, nullptr, strInfoLog);
+            //errorF("\n>>> Shader compiler errors:\n%s", strInfoLog);
+        }
+    }
+
+    static void linkProgram(const GLuint program)
+    {
+        glLinkProgram(program);
+        checkGLError(__FILE__, __LINE__);
+
+        GLint status;
+        glGetProgramiv(program, GL_LINK_STATUS, &status);
+        checkGLError(__FILE__, __LINE__);
+
+        if (status == GL_FALSE)
+        {
+            GLchar strInfoLog[1024] = { 0 };
+            glGetProgramInfoLog(program, sizeof(strInfoLog) - 1, nullptr, strInfoLog);
+            //errorF("\n>>> Program linker errors:\n%s", strInfoLog);
+        }
+    }
+
+    // The "model-view-projection" matrix for the scene.
+    // In this demo, it consists of the camera's view and projection matrices only.
+    math::float4x4 mvpMatrix;
+    unsigned width, height;
+
+private:
+
+    GLuint linePointProgram;
+    GLint  linePointProgram_MvpMatrixLocation;
+
+    GLuint textProgram;
+    GLint  textProgram_GlyphTextureLocation;
+    GLint  textProgram_ScreenDimensions;
+
+    GLuint linePointVAO;
+    GLuint linePointVBO;
+
+    GLuint textVAO;
+    GLuint textVBO;
+
+    static const char* linePointVertShaderSrc;
+    static const char* linePointFragShaderSrc;
+
+    static const char* textVertShaderSrc;
+    static const char* textFragShaderSrc;
+
+
+}; // class IDebugDrawOpenGLImplementation
+
+// ========================================================
+// Minimal shaders we need for the debug primitives:
+// ========================================================
+
+const char* IDebugDrawOpenGLImplementation::linePointVertShaderSrc = "\n"
```

Comment:
Maybe we could move this shaders into a file

Reply:
Cool!